lib/vdsm/storage/lvm.py

```diff
@@ -867,29 +872,44 @@ def getPvs(self, vgName):
         return pvs
 
     def getVg(self, vgName):
-        # Get specific VG
+        """
+        Get specific VG.
+        Raise a VolumeGroupDoesNotExist for LVM command errors or missing VG.
+        """
         vg = self._vgs.get(vgName)
         if not vg or vg.is_stale():
             self.stats.miss()
             vgs = self._reloadvgs(vgName)
+            if vgName not in vgs:
+                raise se.VolumeGroupDoesNotExist(vg_name=vgName)
```

Comment:
I have done the refactor, trying to minimize code duplication. I have done it in a separate commit, but still not the easiest to read for review.

It really allows it to have clear behavior with different function for single- and multi- VG reloading. It will be nicer also when we extend this strategy to LVs and PVs.